Fix bug 1294737: Recognize string format named placeables

MikkCZ opened the pull request.

jotes commented:
@MikkCZ  This looks awesome -> I think we could add some tests.

e.g. Look here:
https://github.com/mozilla/pontoon/blob/master/pontoon/base/tests/test_placeables.py

Feel free to ask any questions -> I should be around today :-) Optionally you're always welcomed at #pontoon@irc.mozilla.org

MikkCZ commented:
Yay! Thank you, I will add the tests tonight.

MikkCZ commented:
I have added a test now and also changed the string compilation to be case insensitive.

In the tests I noticed, they are missing any case testing https://github.com/mozilla/pontoon/commit/a61a5cfbfb6d08dbf46d7c1948051bf7570bc41a. Should I add and commit under this bug?

jotes commented on pontoon/base/utils.py:
```diff
@@ -69,6 +69,13 @@ class SpacesPlaceable(base.Ph):
+class PythonFormatNamedPlaceable(base.Ph):
+    """Placeable handling named format string in python"""
+    istranslatable = False
+    regex = re.compile(r'%\([[\w\d\!\.,\[\]%:$<>\+\-= ]*\)[s|d]', re.IGNORECASE)
```
I think there's a missing %f

jotes commented:
I think you can -> just add a comment.

jotes commented:
You have my humble r+wc :dancer: Good job @MikkCZ !

MikkCZ commented:
Good link, I will rewrite the regex accordingly. Thanks @jotes !

MikkCZ commented:
@jotes r?

jotes commented:
r+ :D I think we can safely pass this to @mathjazz.

mathjazz commented:
Thanks for the patch, @MikkCZ! r+ :-)

Please [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) commits and we're ready to merge & deploy!

MikkCZ commented:
Squashed and rebased.

mathjazz commented:
Deployed.
